modalities/dom/xen-pwa/pwa/service-worker.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license
+ * Copyright (c) 2018 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import Xen from '../../components/xen/xen.js';
+
+class ServiceWorker extends Xen.Base {
+  static get observedAttributes() {
+    return ['href', 'scope', 'disabled'];
+  }
+  _update({href, scope, disabled}, state) {
+    if (!state.registered && !disabled && href) {
+      state.registered = true;
+      this.register(href, scope || '.');
+    }
+  }
+  register(href, scope) {
+    if ('serviceWorker' in navigator) {
+      navigator.serviceWorker.register(href, {scope: scope});
+    }
+  }
+}
+customElements.define('service-worker', ServiceWorker);
```

Comment:
This looks like a copy of the same ServiceWorker class in `modalities` dir above, is it intentional?

Reply:
Good catch. I wasn't sure how to handle it. 

I want pwa to be self-contained but I also want to share service-worker.js.